File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int buy=prices[0];
        int sell=0;

        for(int i=1;i<(int)prices.size();i++){
            if(prices[i]<buy){
                buy=prices[i];
            }else{
                sell=std::max(sell,prices[i]-buy);
            }
        }
        return sell;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
            let buy=prices[0];
        let sell=0;
        for(let i=1;i<prices.length;i++){
            if(prices[i]<buy){
                buy=prices[i]
            }else{
                sell=Math.max(sell,prices[i]-buy)
            }
            console.log(buy,sell)

        }
        return sell
};